src/App.tsx

```diff
@@ -15,36 +15,87 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None = '',
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>([...goodsFromServer]);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const sortGoods = (type: SortType) => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (type === SortType.Alphabetically) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    }
+
+    if (type === SortType.Length) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortType(type);
+  };
```

Comment:
This function directly manipulates the DOM by setting the sorted goods. [CODE KNOWLEDGE] - Instead of directly manipulating the DOM, consider setting state and letting React handle the DOM updates.